addons/survey/security/survey_security.xml

```diff
@@ -97,24 +80,18 @@
             <field name="perm_read" eval="1"/>
             <field name="perm_create" eval="1"/>
         </record>
-        <record id="survey_question_answer_rule_survey_user_read" model="ir.rule">
-            <field name="name">Survey question answer: officer: read all</field>
+        
+        <record id="survey_question_answer_rule_survey_user" model="ir.rule">
+            <field name="name">Survey question answer: officer: assigned or public survey, standalone</field>
             <field name="model_id" ref="survey.model_survey_question_answer"/>
-            <field name="domain_force">[(1, '=', 1)]</field>
-            <field name="groups" eval="[(4, ref('group_survey_user'))]"/>
-            <field name="perm_unlink" eval="0"/>
-            <field name="perm_write" eval="0"/>
-            <field name="perm_read" eval="1"/>
-            <field name="perm_create" eval="0"/>
-        </record>
-        <record id="survey_question_answer_rule_survey_user_cw" model="ir.rule">
-            <field name="name">Survey question answer: officer: create/write/unlink linked to own survey only</field>
-            <field name="model_id" ref="survey.model_survey_question_answer"/>
-            <field name="domain_force">['|', ('question_id.survey_id.create_uid', '=', user.id), ('matrix_question_id.survey_id.create_uid', '=', user.id)]</field>
+            <field name="domain_force">['|', '|', ('question_id.survey_id.user_ids', 'in', user.id),
+                 ('matrix_question_id.survey_id.user_ids', 'in', user.id),
```

Comment:
we don't allow if `matrix_question_id.survey_id.user_ids == False` ? (like we do for `question_id.survey_id.user_ids`)